Formula/w/wget2.rb

```diff
@@ -1,10 +1,16 @@
 class Wget2 < Formula
   desc "Successor of GNU Wget, a file and recursive website downloader"
   homepage "https://gitlab.com/gnuwget/wget2"
-  url "https://gitlab.com/gnuwget/wget2/uploads/83752270de83e103306576e67a1c7c80/wget2-2.0.1.tar.gz"
-  sha256 "0bb7fa03697bb5b8d05e1b5e15b863440826eb845874c4ffb5e32330f9845db1"
+  url "https://ftp.gnu.org/gnu/wget/wget2-2.1.0.tar.gz"
+  sha256 "a05dc5191c6bad9313fd6db2777a78f5527ba4774f665d5d69f5a7461b49e2e7"
   license "GPL-3.0-or-later"
 
+  livecheck do
+    url "https://ftp.gnu.org/gnu/wget/"
+    strategy :gnu
+    regex(%r{href=.*?wget2-(\d+(?:\.\d+)*)(?:\.[a-z]+|/)}i)
```

Comment:
```suggestion
    regex(/href=.*?wget2[._-]v?(\d+(?:\.\d+)+)\.t/i)
```

We only use the `#strategy` method when A) we need to use a strategy other than the one that livecheck uses for the URL or B) we need to use a `strategy` block. In this case, livecheck already uses the `Gnu` strategy for this URL, so this `#strategy` call is unnecessary.

I've added a modified regex above that aligns with our typical regexes for matching a file from an `href` attribute on an HTML page. Namely:

* We use `[._-]` as the delimiter between the name (`wget2`) and the version.
* We use `v?(\d+(?:\.\d+)+)` as the standard regex for versions like `1.2.3`/`v1.2.3`. We only use the looser `*` format when we specifically need to match a version that is usually like `1.2` but can also be `1` (this is rare).
* We typically use a loose `\.t` as the trailing anchor for tarballs.

Reply:
Thank you! I'll adopt your suggestion.

The regex I use is from Homebrew (the one used in Gnu strategy) with some simplification, so I am sure it will be beneficial to apply your change to brew too.

Reply:
> The regex I use is from Homebrew (the one used in Gnu strategy) with some simplification, so I am sure it will be beneficial to apply your change to brew too.

The `Gnu` strategy's default regex has to be somewhat loose in places to match a wider variety of potential scenarios. In a `livecheck` block, we cater the regex to the context of the source we're checking, so it's more specific than the default.

We already use `[._-]` as the aforementioned delimiter in the `Gnu` strategy and we can't make guarantees around the version format or file extension, so the existing strategy regex is still probably fine (unless/until we identify areas for improvement across all formulae using the `Gnu` strategy). I appreciate the suggestion, though.